lib/at_host/at_host.c

```diff
@@ -192,17 +192,32 @@ static void isr(const struct device *dev, void *user_data)
 	}
 }
 
-static int at_uart_init(char *uart_dev_name)
+static int at_uart_init(enum select_uart uart_id)
 {
 	int err;
 	uint8_t dummy;
 
-	uart_dev = device_get_binding(uart_dev_name);
-	if (uart_dev == NULL) {
-		LOG_ERR("Cannot bind %s\n", uart_dev_name);
+	/* Get the UART device */
+	switch (uart_id) {
+	case UART_0:
+		uart_dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(uart0));
+		break;
+	case UART_1:
+		uart_dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(uart1));
+		break;
+	case UART_2:
+		uart_dev = DEVICE_DT_GET_OR_NULL(DT_NODELABEL(uart2));
+		break;
+	default:
+		LOG_ERR("Unknown UART instance %d", uart_id);
 		return -EINVAL;
 	}
 
+	if (uart_dev == NULL) {
+		LOG_ERR("Could not bind UART device");
+		return -EFAULT;
+	}
+
```

Comment:
I don't think this is the right solution, see https://github.com/nrfconnect/sdk-nrf/pull/8260